management/virtualmachine/client.go

```diff
@@ -38,25 +38,24 @@ func (self VirtualMachineClient) CreateDeployment(role Role, cloudServiceName st
 	return self.client.SendAzurePostRequest(requestURL, data)
 }
 
-func (self VirtualMachineClient) GetDeployment(cloudServiceName, deploymentName string) (*DeploymentResponse, error) {
+func (self VirtualMachineClient) GetDeployment(cloudServiceName, deploymentName string) (DeploymentResponse, error) {
+	deployment := DeploymentResponse{}
```

Comment:
declare with `var` if you're not gonna set any fields. (sorta convention and easier to read)